networkx/convert.py

```diff
@@ -78,14 +78,14 @@ def to_networkx_graph(data, create_using=None, multigraph_input=False):
                 create_using=create_using,
                 multigraph_input=data.is_multigraph(),
             )
-            if hasattr(data, "graph"):  # data.graph should be dict-like
-                result.graph.update(data.graph)
-            if hasattr(data, "nodes"):  # data.nodes should be dict-like
```

Comment:
Is it possible to have a Graph object that doesn't have `graph` and `nodes` attributes? The [documentation on subclassing](https://networkx.org/documentation/latest/reference/classes/graph.html#Subclasses_(Advanced)) makes it seem like subclasses must always have these attributes. If that's the case then I think these `if` s can be removed. If not, then we can use the counter example in the tests!

Reply:
I think these checks allow someone to create a class that only holds the adj attribute info and doesn't include node attributes or graph attributes. I don't think that is likely to happen -- probably never has happened.
This change looks good to me!